test/metabase/sync/sync_metadata/fields/sync_metadata_test.clj

```diff
@@ -41,13 +43,33 @@
              :database-type     "Integer"
              :base-type         :type/Integer
              :database-position 0
-             :database-required false}
+             :database-required false
+             :database-auto-incremented false}
             {:name              "My Field"
              :database-type     "Integer"
              :base-type         :type/Integer
              :id                1
              :database-position 0
-             :database-required true})))))
+             :database-required true
+             :database-auto-incremented false})))))
+
+(deftest database-auto-incremented-changed-test
```

Comment:
new test

Reply:
I think this is missing a test case for `database-auto-incremented=nil` in the new-metadata-from-sync arg. But if you change `jdbc-fields-metadata` to always return a non-nil `database-auto-incremented` value (and test that too), then you wouldn't need to write the test for the nil case here.

Reply:
I think you're mistaken because `database-auto-incremented` is not nullable.